geoips/geoips_utils.py

```diff
@@ -445,3 +449,129 @@ def merge_nested_dicts(dest, src, in_place=True):
+            # package.
+            else:
+                log.interactive(f"No '{pkg_name.title()}' Commands were found.")
+
+    except FileNotFoundError:
+        log.interactive(
+            f"No pyproject.toml file found at {toml_path}. Please create one to move "
+            "forward."
+        )
+
+
+def _get_pyproj_scripts(pyproj):
+    """Get command scripts out of the provided pyproject.toml file.
+
+    Parameters
+    ----------
+    pyproj: dict
+        - The pyproject.toml file represented as a dictionary.
+    """
+    try:
+        # Try to access the scripts via poetry's representation
+        scripts = pyproj["tool"]["poetry"]["scripts"]
+    except KeyError:
+        # If that didn't work, they don't exist or it's setuptools-based.
+        # Try to access scripts via setuptools representation.
+        try:
+            scripts = pyproj["project"]["entry-points"]["console_scripts"]
+        except KeyError:
+            # Scripts were not specified, return None
+            scripts = None
+    return scripts
+
+
+def _get_pkg_name_and_logger(pkg_name, provided_log):
+    """Retrieve the Package Name and Logger from the provided variables.
+
+    If pkg_name is None, retrieve pkg_name from the commandline arguments
+    (defaults to 'geoips'). If log is None, set log to the LOG attribute found in this
+    module.
+
+    Parameters
+    ----------
+    pkg_name: str
+        - The name of the GeoIPS Plugin Package whose command's will be exposed.
+        - If None, assume this was called via the commandline and retrieve package_name
+          via that manner. Otherwise use the supplied package_name.
+    provided_log: logging.Logger
+        - If not None, use this log so we can test the output of this function for
+          testing purposes, otherwise retrieve the LOG attribute from this module.
+
+    Returns
+    -------
+    pkg_name, log: str, logging.Logger
+        - The name of the package to retrieve commands from and the logger used to
+          output it.
+    """
+    plugin_packages = [
+        str(ep.value) for ep in get_entry_point_group("geoips.plugin_packages")
+    ]
+    if provided_log:
+        log = provided_log
+    else:
+        log = LOG
+    if pkg_name is None:
+        # This function was called via the command line.
+        argparser = argparse.ArgumentParser("expose command")
+        argparser.add_argument(
+            "--package_name",
+            "-p",
+            type=str.lower,
+            default="geoips",
+            choices=plugin_packages,
+            help="GeoIPS Plugin package to expose.",
+        )
+        ARGS = argparser.parse_args()
+        pkg_name = ARGS.package_name
+    else:
+        # This function was called via python
+        if pkg_name not in plugin_packages:
+            raise PackageNotFoundError(
+                f"No such package named '{pkg_name}' found. Make sure that package is "
```

Comment:
must it be via pip?

Reply:
I think so! Assuming that's how [get_entry_point_group](https://github.com/NRLMMD-GEOIPS/geoips/blob/025c909be866185499aa91fabb0e4cebc3eed61e/geoips/geoips_utils.py#L30) works.

Reply:
See the second paragraph of [importlib.metadata documentation](https://docs.python.org/3/library/importlib.metadata.html). It specifies that it's via ``pip``.

Reply:
It says `importlib.metadata operates on third-party distribution packages installed into Python’s site-packages directory via tools such as [pip](https://pypi.org/project/pip/)`, but not all systems use pip as the manager for python. Eg. arch uses `pacman`. More info: https://packaging.python.org/en/latest/specifications/externally-managed-environments/#externally-managed-environments

_sometimes_ things do need to be via pip (because the software will detect them/update them via pip), and other times they just need to be installed by something that puts them in the right dir

Reply:
^ @evrose54 :)